app/javascript/components/visualization/ScatterPlot.js

```diff
@@ -81,37 +81,50 @@ function RawScatterPlot({
     setHiddenTraces(newShownTraces)
   }
 
-  /** Process scatter plot data fetched from server */
-  function processScatterPlot(clusterResponse=null) {
-    let [scatter, perfTimes] =
-      (clusterResponse ? clusterResponse : [scatterData, null])
-
+  /** Get new, updated scatter object instance, and new layout */
+  function updateScatterLayout(scatter=null) {
+    if (!scatter) {
+      scatter = Object.assign({}, scatterData) // New instance forces render
+    }
     const widthAndHeight = getScatterDimensions(scatter, dimensionProps)
     scatter = Object.assign(scatter, widthAndHeight)
+    scatter.layout = getPlotlyLayout(scatter)
+
+    return scatter
+  }
 
-    const layout = getPlotlyLayout(scatter)
+  /** Update layout, without recomputing traces */
+  function resizePlot() {
+    const scatter = updateScatterLayout()
+    Plotly.relayout(graphElementId, scatter.layout)
+    setScatterData(scatter)
+  }
 
+  /** Update legend counts and recompute traces, without recomputing layout */
+  function updateCountsAndGetTraces(scatter) {
```

Comment:
two different verbs in a method name often indicate something is amiss...     I think this is best just inlined in `processScatterPlot`--this doesn't really add any substantial logic to `getPlotlyTraces()`, and so having to follow an extra method call makes it harder to read.

Reply:
Agreed, the complicated name is a code smell.  This partly aimed to fix a misleading name.  Previously, we put a setter (`setCountsByLabel`) in a getter (`getPlotlyTraces`), which seemed worse.

To your latter point: inlining it in `processScatterPlot` would require extra parameters and more complexity in the callers (`processScatterPlot` and a `useUpdateEffect`) to retain these performance optimizations.  That seems like a wash in terms of readability.